sqllogictest-engines/src/mysql.rs

```diff
@@ -0,0 +1,83 @@
+use std::process::Command;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use mysql_async::prelude::FromValue;
+use mysql_async::prelude::Queryable;
+use mysql_async::FromValueError;
+use sqllogictest::{DBOutput, DefaultColumnType};
+
+type Result<T> = std::result::Result<T, mysql_async::Error>;
+
+/// Connection configuration. This is a re-export of [`mysql_async::Opts`].
+pub type MySqlConfig = mysql_async::Opts;
+
+/// MySQL engine based on the client from [`mysql_async`].
+pub struct MySql {
+    pool: mysql_async::Pool,
+}
+
+impl MySql {
+    /// Connects to the MySQL server with the given `config`.
+    pub async fn connect(config: MySqlConfig) -> Result<Self> {
+        let pool = mysql_async::Pool::new(config);
+        Ok(Self { pool })
+    }
+}
+
+#[async_trait]
+impl sqllogictest::AsyncDB for MySql {
+    type Error = mysql_async::Error;
+    type ColumnType = DefaultColumnType;
+
+    async fn run(&mut self, sql: &str) -> Result<DBOutput<Self::ColumnType>> {
+        let mut conn = self.pool.get_conn().await?;
+        if sql.trim_start().to_lowercase().starts_with("select") {
+            let mut output = vec![];
+            let rows: Vec<mysql_async::Row> = conn.query(sql).await?;
+            for row in rows {
+                let mut row_vec = vec![];
+                for i in 0..row.len() {
+                    let value: std::result::Result<String, FromValueError> =
+                        FromValue::from_value_opt(row[i].clone());
+                    match value {
+                        Ok(value) => {
```

Comment:
Have you tested whether values in all types able to be converted into `String`? By inspecting the source code it seems that only `Value::Bytes` will succeed. Shall we use `as_sql` instead?

Reply:
Thanks for your insight. Yes, the conversion here can only succeed when it is Value::Bytes. I didn't notice this before.

In addition, because mysql_async uses the text protocol when using the query API, all database types are encoded using strings and returned as Value::Bytes or Value::NULL on client side. The use of other types of Value is only involved when using the binary protocol, which is how the exec API is used. For this, the [mysql_async document](https://github.com/blackbeam/mysql_async?tab=readme-ov-file#mysql-query-protocols) mentions, and I have also verified it locally.

Based on Value::Bytes, it seems that we can parse it ourselves. I have updated the logic here again. I don't know if it makes sense to you.

Reply:
For as_sql it seems to output some annotations that don't align with existing behavior, such as adding `''` to the output, but it works for all types. I am not sure whether to use this API.

Reply:
> The use of other types of Value is only involved when using the binary protocol, which is how the exec API is used. For this, the [mysql_async document](https://github.com/blackbeam/mysql_async?tab=readme-ov-file#mysql-query-protocols) mentions, and I have also verified it locally.

This is interesting to know. Thanks for sharing.

> For as_sql it seems to output some annotations that don't align with existing behavior, such as adding `''` to the output, but it works for all types. I am not sure whether to use this API.

Agree that this is a problem. Ideally, the same test case should be able to pass using different engines like Postgres vs MySQL.